### DISCUSSION TITLE
Querying Relation with list of ids

### DISCUSSION THREAD

**Discussion by KaufmannDavid** [prisma/prisma#12742] [Q&A]
Hi,
i don't know how to solve the following problem with the prisma client. Lets say I have relations A and B and a one to many relation between them, where one instance of A can be in relation with multiple instances of B. I now have a list of ids of instances of B and want to query A in a way that I get all instances of A which are in relation with all the instances of B from my list. I don't care if the instances of A are also in relation with other instances of B. I hope my problem was clear. Unfortunately I couldn't find a way to solve this maybe you can help me figure one out.

Thanks a lot

**Comment by austincrim** [comment]
Hey @KaufmannDavid 👋,

If I'm understanding your question correctly, I think you can use our [`some` option](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#some) to achieve this.

Here's a concrete example using a `Store` that has many `Products`:
```ts
let productIds = [1, 2, 3]
let result = await prisma.store.findMany({
  where: {
    products: {
      some: {
        id: {
          in: productIds
        }
      }
    }
  }
})
```
This query will find all the `Store`'s that have at least one `Post` with an id of 1, 2, or 3.

Is that the kind of result you wanted? If not, could you provide a concrete example to help me understand?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
